tests/test_dht_experts.py

```diff
@@ -49,12 +48,11 @@ def test_store_get_experts():
 
 
 @pytest.mark.forked
-def test_beam_search(dht_size=20, total_experts=128, batch_size=32, initial_peers=3, beam_size=4, parallel_rpc=4,
+def test_beam_search(n_peers=20, total_experts=128, batch_size=32, beam_size=4, parallel_rpc=4,
                      grid_dims=(32, 32, 32)):
-    dht = []
-    for i in range(dht_size):
-        neighbors_i = [f'{LOCALHOST}:{node.port}' for node in random.sample(dht, min(initial_peers, len(dht)))]
-        dht.append(hivemind.DHT(start=True, initial_peers=neighbors_i, parallel_rpc=parallel_rpc))
+    dht = [hivemind.DHT(start=True)]
+    initial_peers = dht[0].get_visible_maddrs()
+    dht += [hivemind.DHT(initial_peers=initial_peers, start=True) for _ in range(n_peers - 1)]
```

Comment:
Not sure that replacing the logic there is worth it: we want to check that even when connecting to random peers (so that the sets of initial neighbors can be disjoint), each peer can reach any other peer

Reply:
The libp2p version of the DHT starts much longer (> 0.4 sec) since we need to create and connect to the libp2p daemon. Therefore, it is crucial for the majority of tests to start the DHT instances in parallel.

This was not possible with the old topology (a new node connects to a random node among previously started) but is possible with this simpler ("star-shaped") topology.

There are still tests in `test_dht_node.py` that have more complex topology (to test that each peer can reach any other peer). I'd argue that other tests don't need to test this again to save the testing time and follow the unit test philosophy (each component should be tested separately).